src/Utility/Security.php

```diff
@@ -121,17 +121,35 @@ public static function randomBytes(int $length): string
      * Creates a secure random string.
      *
      * @param int $length String length. Default 64.
+     * @param bool $includeSpecialChars If to include special characters. Default false.
      * @return string
      */
-    public static function randomString(int $length = 64): string
+    public static function randomString(int $length = 64, bool $includeSpecialChars = false): string
     {
-        return substr(
-            bin2hex(Security::randomBytes((int)ceil($length / 2))),
+        $string = '';
+
+        if ($length === 0) {
+            return $string;
+        }
+
+        if ($includeSpecialChars) {
+            $special_charset = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~';
+
+            for ($i = 0; $i < rand(1, $length); $i++) {
```

Comment:
I would move the random length generation out of the for loop.

Reply:
Out of the loop in the new version